src/transformStateWithClones.js

```diff
@@ -8,6 +8,29 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
-}
+  const arr = [];
+  const stateNew = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateNew, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateNew[key];
+        }
+        break;
 
+      case 'clear':
+        for (const key in stateNew) {
+          delete stateNew[key];
+        }
```

Comment:
or you could rewrite the stateNew because you don't mutate incoming object and need to return the new obj.
by rewritten you don't do all that iteration in this loop)